tool/tbot/main.go

```diff
@@ -221,6 +251,81 @@ func onStart(botConfig *config.BotConfig) error {
 	return renewLoop(ctx, botConfig, authClient, ident)
 }
 
+func hasTokenChanged(configTokenBytes, identityBytes []byte) bool {
+	if len(configTokenBytes) == 0 || len(identityBytes) == 0 {
+		return false
+	}
+
+	return !bytes.Equal(identityBytes, configTokenBytes)
+}
+
+// checkDestinations checks all destinations and tries to create any that
+// don't already exist.
+func checkDestinations(cfg *config.BotConfig) error {
+	// Note: This is vaguely problematic as we don't recommend that users
+	// store renewable certs under the same user as end-user certs. That said,
+	//  - if the destination was properly created via tbot init this is a no-op
+	//  - if users intend to follow that advice but miss a step, it should fail
+	//    due to lack of permissions
+	storage, err := cfg.Storage.GetDestination()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// TODO: consider warning if ownership of all destintions is not expected.
+
+	if err := storage.Init(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	for _, dest := range cfg.Destinations {
+		destImpl, err := dest.GetDestination()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		if err := destImpl.Init(); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	return nil
+}
+
+// checkIdentity performs basic startup checks on an identity and loudly warns
+// end users if it is unlikely to work.
+func checkIdentity(ident *identity.Identity) error {
+	var validAfter time.Time
+	var validBefore time.Time
+
+	if ident.X509Cert != nil {
+		validAfter = ident.X509Cert.NotBefore
+		validBefore = ident.X509Cert.NotAfter
+	} else if ident.SSHCert != nil {
+		validAfter = time.Unix(int64(ident.SSHCert.ValidAfter), 0)
+		validBefore = time.Unix(int64(ident.SSHCert.ValidBefore), 0)
+	} else {
+		return trace.BadParameter("identity is invalid and contains no certificates")
+	}
+
+	now := time.Now().UTC()
+	if now.After(validBefore) {
+		log.Errorf(
+			"Identity appears to have expired. The renewal is likely to fail. (expires: %s, current time: %s)",
```

Comment:
Nit: I think it's pretty definitive, no? :)
```suggestion
			"Identity has expired. The renewal will fail. (expires: %s, current time: %s)",
```